### PR TITLE
Feature: APP-907 - Proposal Details: Status Widget

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description
- Adds status widget to proposal details page

Task: [APP-907](https://aragonassociation.atlassian.net/browse/APP-907)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/ui-components/src/components/progress/status.tsx [hunk 1]
```diff
@@ -51,8 +52,8 @@ export const ProgressStatus: React.FC<ProgressStatusProps> = ({
   date,
   block,
 }) => {
-  if (mode !== 'upcoming' && !date) date = 'No information available';
-  if (mode === 'upcoming') date = '';
+  if (mode !== 'upcoming' && mode !== 'failed' && !date)
+    date = 'No information available';
```
Comment (sepehr2github) [inline packages/ui-components/src/components/progress/status.tsx:56]: As a reminder, We need to translate this later and avoid translation.

**Review by sepehr2github — APPROVED** [review]
Thanks, Seems good! Just a minor thing

File: packages/ui-components/src/components/progress/status.tsx (hunk 1 above)
Comment (Fabricevladimir) [reply packages/ui-components/src/components/progress/status.tsx:56]: Yuck, yes, you're right.

File: packages/web-app/src/utils/proposals.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-fallthrough */
```
Comment (WissenIstNacht) [inline packages/web-app/src/utils/proposals.ts]: This doesn't seem necessary? Your switch statemets all return.

File: packages/web-app/src/pages/proposal.tsx [hunk 3]
```diff
@@ -280,18 +283,43 @@ const Proposal: React.FC = () => {
...
+      proposal?.creationDate
+    )
+      return getProposalStatusSteps(
+        proposal.status,
+        proposal.startDate,
+        proposal.endDate,
+        proposal.creationDate,
+        '123,123,123',
+        '456,456,456',
+        new Date() // TODO: change to proposal.executionDate from sdk
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx:314]: Because it's not exposed by it yet, I assume?

**Review by WissenIstNacht — APPROVED** [review]
Thanks a lot.

Could you remove the lint rule unless it's ABSOLUTELY necessary? Otherwise, ready to go from my side.

File: packages/web-app/src/pages/proposal.tsx (hunk 3 above)
Comment (Fabricevladimir) [reply packages/web-app/src/pages/proposal.tsx:314]: Not exposed, not available even on subgraph either. Core is aware of that though.

**Pull request merged**